src/main/java/ru/touchin/roboswag/core/observables/collections/changes/CollectionChange.java

```diff
@@ -0,0 +1,40 @@
+package ru.touchin.roboswag.core.observables.collections.changes;
+
+import android.support.annotation.NonNull;
+
+import java.util.Collection;
+
+/**
+ * Class which is representing change of collection. There could be multiple changes applied to collection.
+ */
+public class CollectionChange<TItem> {
+
+    private final int number;
+    @NonNull
+    private final Collection<Change> changes;
+
+    public CollectionChange(final int number, @NonNull final Collection<Change> changes) {
```

Comment:
I think it should be class like:

ChangesCollection{
List oldItems or initialItems
List newItems or modifiedItems
Predicate samePredicate (null by default)

ChangesCollection(oldItems, newItems){
this.oldItems = new ArrayList(oldItems) - it's important to create new list or at least unmodifiable collection
}

getInsertedItems() {
calculate by predicate or by equals if predicate is null
}

getRemovedItems() {
same as above
}

}